pkg/chunked/compression_linux.go

```diff
@@ -48,21 +48,21 @@ func readEstargzChunkedManifest(blobStream ImageSourceSeekable, blobSize int64,
 		Offset: uint64(blobSize - footerSize),
 		Length: uint64(footerSize),
 	}
-	parts, errs, err := blobStream.GetBlobAt([]ImageSourceChunk{chunk})
+
+	footer := make([]byte, footerSize)
```

Comment:
IIRC, a lot of the code could benefit from a single “ask for one chunk, read it into memory, validate against a digest” utility.

Arguably that’s a cleanup that should not block this PR…